main.js

```diff
@@ -10,43 +11,70 @@ import {
   getAssetSizes,
 } from './lib/helpers';
 
-let octokit;
+async function getActionInputs() {
+  const workingDirectory = getInput('working-directory', { required: false });
+  const usePrArtifacts = getInput('use-pr-artifacts', { required: false });
+  const token = getInput('repo-token', { required: true });
+  
+  const cwd = path.join(process.cwd(), workingDirectory);
+  debug(`cwd: ${cwd}`);
+  debug(`token: ${token}`);
 
-async function run() {
-  try {
-    const myToken = getInput('repo-token', { required: true });
-    octokit = new GitHub(myToken);
-    const pullRequest = await getPullRequest(context, octokit);
+  return { token, cwd, usePrArtifacts };
+}
 
-    const prAssets = await getAssetSizes();
+async function diffAssets({ pullRequest, cwd, usePrArtifacts }) {
+  const prAssets = await getAssetSizes({ cwd, build: !usePrArtifacts });
 
-    await exec(`git checkout ${pullRequest.base.sha}`);
+  await exec(`git checkout ${pullRequest.base.sha}`, { cwd });
 
-    const masterAssets = await getAssetSizes();
+  const masterAssets = await getAssetSizes({ cwd, build: true });
 
-    const fileDiffs = diffSizes(normaliseFingerprint(masterAssets), normaliseFingerprint(prAssets));
+  const fileDiffs = diffSizes(
+    normaliseFingerprint(masterAssets),
+    normaliseFingerprint(prAssets),
+  );
 
-    const body = buildOutputText(fileDiffs);
 
-    try {
-      await octokit.issues.createComment({
-        owner: context.repo.owner,
-        repo: context.repo.repo,
-        issue_number: pullRequest.number,
-        body,
-      });
-    } catch (e) {
-      console.log(`Could not create a comment automatically. This could be because github does not allow writing from actions on a fork.
+  return fileDiffs;
+}
+
+async function commentOnPR({ octokit, pullRequest, fileDiffs }) {
+  const body = buildOutputText(fileDiffs);
+
+  try {
+    await octokit.issues.createComment({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      issue_number: pullRequest.number,
+      body,
+    });
+  } catch (e) {
+    console.error(e);
+    console.log(`Could not create a comment automatically. This could be because github does not allow writing from actions on a fork.
 
 See https://github.community/t5/GitHub-Actions/Actions-not-working-correctly-for-forks/td-p/35545 for more information.`);
 
-      console.log(`Copy and paste the following into a comment yourself if you want to still show the diff:
+    console.log(`Copy and paste the following into a comment yourself if you want to still show the diff:
 
 ${body}`);
-    }
+  }
+}
+
+
+export default async function run() {
+  try {
+    const { token, cwd, usePrArtifacts } = getActionInputs();
```

Comment:
I’m trying this out on an Ember application that lives in a subdirectory and I was getting authentication failures because this call needs an `await` 😳